ocaml/quicktest/qt_filter.ml

```diff
@@ -116,36 +116,41 @@ module SR = struct
     ; required_sm_api_version
     }
 
-  let list_selected_srs rpc session_id =
-    Client.Client.SR.get_all rpc session_id
-    |> List.filter
-      (fun sr ->
-         Client.Client.SR.get_PBDs rpc session_id sr
-         |> List.exists
-           (fun pbd -> Client.Client.PBD.get_currently_attached rpc session_id pbd)
-      )
-    |> List.map (get_sr_info rpc session_id)
-
-  let only sr () = [get_sr_info !A.rpc !session_id sr]
-
-  let all_srs =
-    let is_shared_or_local sr_info =
+  let list_srs_connected_to_localhost rpc session_id =
+    let is_shared_or_local sr =
       (** allows us to ignore srs which don't belong to us.
         * this remedies some failing tests if we run the quicktest binary on a host in a pool *)
-      let sr = sr_info.Qt.sr in
-      let session_id = !session_id in
-      let rpc = !A.rpc in
       let pbds = Client.Client.SR.get_PBDs ~rpc ~session_id ~self:sr in
+      let does_pbd_belong_to_me pbd =
+        let pbd_host = Client.Client.PBD.get_host ~rpc ~session_id ~self:pbd in
+        let pbd_host_uuid = Client.Client.Host.get_uuid ~rpc ~session_id ~self:pbd_host in
+        pbd_host_uuid = A.localhost_uuid
+      in
       match pbds with
-      | [pbd] -> begin
-          let pbd_host = Client.Client.PBD.get_host ~rpc ~session_id ~self:pbd in
-          let pbd_host_uuid = Client.Client.Host.get_uuid ~rpc ~session_id ~self:pbd_host in
-          pbd_host_uuid = A.localhost_uuid (* sr is not shared, test it only if it is ours *)
-        end
+      | [pbd] -> (* sr is not shared, test it only if it is ours *)
+        does_pbd_belong_to_me pbd
       | []    -> Printf.eprintf "SR (%s) has no pbds %s" (Ref.string_of sr) __LOC__; false
-      | _::_  -> true (* sr is shared, and we can see it *)
+      | _::_ as pbds  -> (** sr is shared, so we have access to is
```

Comment:
```
| [] ->
| [pbd] ->
| pbds -> when List.exists does_pbd_belong_to_me pbds ->
| _  ->
```
I would have expected `List.for_all` rather than `List.exists` but maybe knowing one is good enough.

Reply:
`for_all` is not right - each host in the pool should have a PBD for any shared SR